libpod/container_internal_linux.go

```diff
@@ -698,21 +698,37 @@ func (c *Container) makeBindMounts() error {
 			// If it doesn't, don't copy them
 			resolvPath, exists := bindMounts["/etc/resolv.conf"]
 			if exists {
-
 				c.state.BindMounts["/etc/resolv.conf"] = resolvPath
 			}
+
+			// check if dependency container has an /etc/hosts file
 			hostsPath, exists := bindMounts["/etc/hosts"]
-			if exists {
-				c.state.BindMounts["/etc/hosts"] = hostsPath
+			if !exists {
+				return errors.Errorf("error finding hosts file of dependency container %s for container %s", depCtr.ID(), c.ID())
+			}
+
+			depCtr.lock.Lock()
```

Comment:
I think we can drop the lock here. We only really want to read the hosts file; If we fail to open it, we get a sane enough error back, if we succeed, we have a good file descriptor and can read the whole thing, even if the container is stopped

Reply:
I just figured generateHosts assumed a lock. What if two containers are added at the same time and they both try to append to the fd at the same time? odd case, but I think it's safer to have the lock. i'll leave it up to you.

Reply:
Good point on the parallel writes - yeah, safest to leave it, then.